spec/swift/app_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative "../spec_helper"
+
+describe "App" do
+  include Rack::Test::Methods
+
+  def app
+    LiquorCabinet
+  end
+
+  it "returns 404 on non-existing routes" do
+    get "/virginmargarita"
+    last_response.status.must_equal 404
+  end
+
+  describe "PUT requests" do
+    context "authorized" do
+      before do
+        redis.sadd "authorizations:phil:amarillo", [":rw"]
+        header "Authorization", "Bearer amarillo"
+      end
+
+      it "creates the metadata object in redis" do
+        put_stub = OpenStruct.new(headers: {etag: "bla"})
+        RemoteStorage::Swift.stub_any_instance :has_name_collision?, false do
+          RestClient.stub :put, put_stub do
+            put "/phil/food/aguacate", "si"
+          end
+        end
+
+        metadata = redis.hgetall "rs_meta:phil:food/aguacate"
```

Comment:
Looks like these are not deleted after test runs, so the test might actually work although writing failed.